js/app/warehouse.js

```diff
@@ -32,10 +32,29 @@ var Warehouse = new Knowledge.Class({
                     return current_dir[file_name][module_path];
                 }
                 return current_dir[file_name];
-            }, imports.app.modules);
-        } catch (error if (error.message.startsWith('No JS module'))) {
```

Comment:
I would just move the `if` statement inside the catch clause, because otherwise this will swallow other errors:
```js
} catch (error) {
    if (error.message.startsWith('No JS module'))
        return null;
    throw error;
}
```